server/src/com/cloud/api/ApiResponseHelper.java

```diff
@@ -479,6 +479,17 @@ public SnapshotResponse createSnapshotResponse(Snapshot snapshot) {
             if (zone != null) {
                 snapshotResponse.setZoneId(zone.getUuid());
             }
+
+            if (volume.getVolumeType().equals(Volume.Type.ROOT)) {
+                VMInstanceVO instance = ApiDBUtils.findVMInstanceById(volume.getInstanceId());
+                if (instance != null) {
+                    GuestOS guestOs = ApiDBUtils.findGuestOSById(instance.getGuestOSId());
```

Comment:
My mistake. I guess it would only return the ID of the Guest OS... so from the interface level it'd be from volume ID -> Guest OS ID?